### PR TITLE
DISCL-207: Resize and zoom around point from gesture or mouse event

### PR DESCRIPTION
**Pull request by tribal-tec**
Change-Id: I38bc4e05f291b8c39ae85448621e1861dc90bf35

### CODE REVIEW

**Comment by tribal-tec** [comment]
Retest this please

File: tests/cpp/core/ContentWindowTests.cpp [hunk 1]
```diff
@@ -137,55 +137,14 @@ BOOST_AUTO_TEST_CASE( testZoom )
...
-
-    window.setZoom( 1.0 );
-    BOOST_CHECK_EQUAL( window.getZoomCenter().x(), zoomCenter.x() );
-    BOOST_CHECK_EQUAL( window.getZoomCenter().y(), zoomCenter.y() );
+    BOOST_REQUIRE( window.getZoomRect() == QRectF( 0.0, 0.0, 1.0, 1.0 ));
+
+    window.setZoomRect( QRectF( 0.2, 0.2, 0.7, 0.7 ));
+    BOOST_CHECK( window.getZoomRect() == QRectF( 0.2, 0.2, 0.7, 0.7 ));
+    window.setZoomRect( QRectF( -0.1, 0.2, 0.7, 0.7 ));
+    BOOST_CHECK( window.getZoomRect() == QRectF( -0.1, 0.2, 0.7, 0.7 ));
```
Comment (rdumusc) [inline tests/cpp/core/ContentWindowTests.cpp]: can we allow this, or should the window itself constrain the zoom rect? Is there no risks when rendering with?

**Comment by rdumusc** [comment]
+1, good changes!

File: tests/cpp/core/ContentWindowTests.cpp (hunk 1 above)
Comment (tribal-tec) [inline tests/cpp/core/ContentWindowTests.cpp]: I'm also struggling with this. This way, the window takes any rectangle, and the constrains are applied by the zoomInteractionDelegate which is fine in the context of the application, but it feels a bit strange in the test.
Rendering is strange, but not risky. It does not happen anyways as the zoom rectangle is controlled/constrained by the delegate. The test just states that the window does not change anything in the rectangle.

Comment (rdumusc) [inline tests/cpp/core/ContentWindowTests.cpp]: OK, all good then! And maybe certain content types could even decide to handle negative zoom factor in their own manner...

**Pull request merged**